### DISCUSSION TITLE
Is it common that everyone is ignoring that the scroll position does not gets restored when switching tabs?

### DISCUSSION THREAD

**Discussion by kasem-sm** [arkivanov/Decompose#629] [Q&A]
I was checking out samples such as [Confetti](https://github.com/joreilly/Confetti), [TvManiac](https://github.com/thomaskioko/tv-maniac) & the official sample (Article List/Multi Pane) and when I switch to other tab, the scroll position is lost. 

In one of my app, I am extending my presenter with `InstanceKeeper.Instance` but seems like it does not helps saving state.

**Comment by arkivanov** [comment]
If we speak about Compose UI, the scroll position (and the entire UI state) should be preserved, as long as you use `bringToFront` for tab screens and `Children` function to display the screens. I'm pretty sure Confetti worked fine a while ago, not sure about other samples. The idea is that `bringToFront` keeps inactive screens in the stack, and the `Children` function is able to restore the state later.

cc: @joreilly

**Comment by kasem-sm** [comment]
If the screen is already in stack, will calling bringToFront recreate it? It seems to do so

**Reply by arkivanov** [reply]
bringToFront will recreate the component only if the configuration has changed. If it's `equal` - then the instance is just brought to front.
